src/main/java/duke/command/IndexedCommand.java

```diff
@@ -0,0 +1,13 @@
+package duke.command;
+
+abstract class IndexedCommand extends Command{
+    private final int index;
+    public IndexedCommand(int position){
+        this.index = position - 1;
+    }
+
+    @Override
+    public String[] run() {
```

Comment:
Should this method be named with a more suiting name? Perhaps a name that explains why this method has to return an array of Strings?